libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/EHorizon.kt

```diff
@@ -99,20 +98,20 @@ class EHorizon internal constructor(
             ")"
     }
 
-    private fun mpp(start: Edge): List<List<Edge>> {
+    private fun mpp(start: EHorizonEdge): List<List<EHorizonEdge>> {
         // if it is not MPP return empty list
         if (!start.isMpp()) return emptyList()
         if (start.out.isEmpty()) return listOf(listOf(start))
-        val result = LinkedList<LinkedList<Edge>>()
+        val result = LinkedList<LinkedList<EHorizonEdge>>()
```

Comment:
Probably worth using `ArrayList` for result instead. AFAIR `LinkedList.get` has O(N) complexity.

Reply:
working on the refactoring of the whole method

Reply:
fixed